server/src/main/java/org/opensearch/Version.java

```diff
@@ -363,7 +285,14 @@ private static Version fromStringSlow(String version) {
     public final org.apache.lucene.util.Version luceneVersion;
 
     Version(int id, org.apache.lucene.util.Version luceneVersion) {
-        this.id = id;
+        // flip the 28th bit of the ID; identify as an opensearch vs legacy system:
+        // we start from version 1 for opensearch, so ignore the 0 (empty) version
+        if(id != 0) {
+            this.id = id ^ MASK;
+            id &= 0xF7FFFFFF;
+        } else {
```

Comment:
should there be an else case?

Reply:
It's to support `V_EMPTY` but it looks like that's not even used anywhere. So we could probably remove it. I'd prefer doing that in a separate PR?

Reply:
Sounds good. Thanks!

Reply:
And good catch! I'll open an issue.